src/libpmemobj/tx.c

```diff
@@ -1116,6 +1178,99 @@ tx_realloc_common(struct tx *tx, PMEMoid oid, size_t size, uint64_t type_num,
 }
 
 /*
+ * pmemobj_tx_ctx_init -- initializes the transaction context
+ */
+void
+pmemobj_tx_ctx_init(void)
+{
+	LOG(3, NULL);
+	int result = pthread_once(&tx_ctx_once, tx_ctx_create_key);
```

Comment:
Can't we just use __thread variable, just like before?

The ctx WILL be NULL if the application closes properly (because it had to call tx_end()), so we don't need the destructor.